Cython/Compiler/ExprNodes.py

```diff
@@ -13192,6 +13200,16 @@ def analyse_types(self, env):
 
     def generate_result_code(self, code):
         code.putln('%s; %s' % (
+            self.arg.type.to_py_call_code(
```

Comment:
Can you ever end up with `arg` as something that isn't a `SliceIndexNode`? For example:

    a = some_array[1:10]
    return a

Could `a` pick up the type from the indexing operation, but the coercion would happen to a `NameNode`?

Reply:
I little bit forgot what is there, I know that I did it by potentially wrong way, comments will help to understand where are errors